### PR TITLE
Make workload descriptions localizable

### LINKED ISSUE #20922: Given manifest layout ManifestReader should read out localized description
sample attachement
[microsoft.net.workload.mono.toolchain.zip](https://github.com/dotnet/sdk/files/7146668/microsoft.net.workload.mono.toolchain.zip)

### PR DESCRIPTION
**Pull request by mhutch**
Currently missing tests - will add.

### CODE REVIEW

File: src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadManifestReader.Localization.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+namespace Microsoft.NET.Sdk.WorkloadManifestReader
+{
+    partial class WorkloadManifestReader
+    {
+        private static LocalizationCatalog ReadLocalizationCatalog(ref Utf8JsonStreamReader reader)
+            => new LocalizationCatalog(ReadStringDictionary(ref reader));
+
+        public static string? GetLocalizationCatalogFilePath(string manifestFilePath, CultureInfo? culture = null)
+        {
+            culture ??= CultureInfo.CurrentUICulture;
```
Comment (wli3) [inline src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadManifestReader.Localization.cs:27]: according to https://github.com/dotnet/sdk/blob/a30e465a2e2ea4e2550f319a2dc088daaafe5649/src/Cli/dotnet/UILanguageOverride.cs#L27

we should read DefaultThreadCurrentUICulture

**Comment by wli3** [comment]
the implemetation looks good

**Comment by wli3** [comment]
@mhutch what is the timeline adding the tests?

**Comment by wli3** [comment]
RC2 is closing in a week

File: src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadManifestReader.Localization.cs (hunk 1 above)
Comment (mhutch) [reply src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadManifestReader.Localization.cs]: A thread will use that if it does not have an explicit culture set: https://github.com/dotnet/runtime/blob/cd1b4cff818a700b6ff82fde024c65b036979894/src/libraries/System.Private.CoreLib/src/System/Globalization/CultureInfo.cs#L394

**Review by sfoslund — APPROVED** [review]

**Review by sfoslund — APPROVED** [review]

**Comment by sfoslund** [comment]
Approved by Don, merging..

**Pull request merged**